### PR TITLE
Add support for openEuler 22.03

### PR DESCRIPTION
**Pull request by martin-g**
Adds logic to support building and testing on openEuler 22.03.

For the time being it uses urls to openEuler's OBS and Github repo to fetch ohpc-release.oe2203.rpm until this package is made available at http://obs.openhpc.community

Signed-off-by: Yikun Jiang <yikunkero@gmail.com>
Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

### CODE REVIEW

**Comment by adrianreber** [comment]
I would prefer smaller steps. Please create a PR just to introduce the Github Actions CI part based on https://obs.openhpc.community/project/show/OpenHPC:3.0:Factory

**Comment by martin-g** [comment]
Done!
All the CI related changes have been moved to https://github.com/openhpc/ohpc/pull/1647!

File: components/admin/powerman/SPECS/powerman.spec [hunk 1]
```diff
@@ -22,7 +22,7 @@ Group: %{PROJ_NAME}/admin
 Url: http://code.google.com/p/powerman/
 Source0: https://github.com/chaos/%{pname}/releases/download/%{version}/%{pname}-%{version}.tar.gz
 
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/admin/powerman/SPECS/powerman.spec]: https://github.com/openhpc/ohpc/pull/1656

File: components/dev-tools/autoconf/SPECS/autoconf.spec [hunk 2]
```diff
@@ -24,7 +24,7 @@ Source0:   https://ftp.gnu.org/gnu/autoconf/autoconf-%{version}.tar.gz
 BuildRequires: m4
 Requires: m4
 
-%if 0%{?rhel_version} || 0%{?centos_version} || 0%{?rhel}
+%if 0%{?rhel_version} || 0%{?centos_version} || 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/dev-tools/autoconf/SPECS/autoconf.spec:29]: https://github.com/openhpc/ohpc/pull/1657

File: components/dev-tools/automake/SPECS/automake.spec [hunk 3]
```diff
@@ -20,6 +20,7 @@ License:   GNU GPL
 Group:     %{PROJ_NAME}/dev-tools
 URL:       http://www.gnu.org/software/automake/
 Source0:   https://ftp.gnu.org/gnu/automake/automake-%{version}.tar.gz
+Patch0:    fix-automake-build-error.patch
```
Comment (Yikun) [inline components/dev-tools/automake/SPECS/automake.spec:23]: out date

File: components/dev-tools/automake/SPECS/automake.spec [hunk 4]
```diff
@@ -28,6 +29,11 @@ BuildRequires: autoconf%{PROJ_DELIM} >= 2.69
 
 BuildRequires: perl(Thread::Queue) make
 
+%if 0%{?openEuler}
```
Comment (Yikun) [inline components/dev-tools/automake/SPECS/automake.spec:34]: Replaced by https://github.com/openhpc/ohpc/commit/2ab677cbaf5b44c1afa4ce462f0c20c71e4eed52 (L30)

File: components/dev-tools/cmake/SPECS/cmake.spec [hunk 5]
```diff
@@ -31,7 +31,7 @@ BuildRequires:  xz-devel
 BuildRequires:  zlib-devel
 BuildRequires:  pkgconfig
 
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/dev-tools/cmake/SPECS/cmake.spec]: https://github.com/openhpc/ohpc/pull/1657

File: components/dev-tools/spack/SPECS/spack.spec [hunk 6]
```diff
@@ -42,7 +42,7 @@ Requires: zstd
 Requires: file
 Requires: git
 Requires: curl
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/dev-tools/spack/SPECS/spack.spec]: https://github.com/openhpc/ohpc/pull/1657

File: components/mpi-families/libfabric/SPECS/libfabric.spec [hunk 7]
```diff
@@ -28,7 +28,7 @@ BuildRequires: gcc make
 Buildrequires: ofed
 BuildRequires: rdma-core-devel infiniband-diags-devel
 %endif
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/mpi-families/libfabric/SPECS/libfabric.spec]: https://github.com/openhpc/ohpc/pull/1659

File: components/mpi-families/mvapich2/SPECS/mvapich2.spec [hunk 8]
```diff
@@ -61,7 +61,7 @@ Conflicts: %{pname}-%{compiler_family}%{PROJ_DELIM}
 Buildrequires: ofed
 BuildRequires: rdma-core-devel infiniband-diags-devel
 %endif
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/mpi-families/mvapich2/SPECS/mvapich2.spec]: https://github.com/openhpc/ohpc/pull/1659

File: components/mpi-families/openmpi/SPECS/openmpi.spec [hunk 9]
```diff
@@ -68,7 +68,7 @@ BuildRequires:  libevent-devel
 %endif
 %if 0%{with_ofi}
 BuildRequires:  libfabric%{PROJ_DELIM}
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/mpi-families/openmpi/SPECS/openmpi.spec]: https://github.com/openhpc/ohpc/pull/1659

File: components/parallel-libs/boost/SPECS/boost.spec [hunk 10]
```diff
@@ -33,7 +33,7 @@ BuildRequires:  python3-numpy-devel
 BuildRequires:  libquadmath0
 %endif
 %else
-# Assume RHEL/Fedora distro
+# Assume RHEL/Fedora/openEuler distro
```
Comment (Yikun) [inline components/parallel-libs/boost/SPECS/boost.spec:36]: https://github.com/openhpc/ohpc/pull/1676

File: components/parallel-libs/mumps/SPECS/mumps.spec [hunk 11]
```diff
@@ -36,7 +36,7 @@ Patch0:         mumps-5.0.1-shared-mumps.patch
 Patch1:         mumps-5.0.0-shared-pord.patch
 Requires:       lmod%{PROJ_DELIM} >= 7.6.1
 
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/parallel-libs/mumps/SPECS/mumps.spec]: https://github.com/openhpc/ohpc/pull/1677

File: components/parallel-libs/ptscotch/SPECS/ptscotch.spec [hunk 12]
```diff
@@ -31,7 +31,7 @@ Patch0:  scotch-6.0.4-destdir.patch
 
 BuildRequires:	flex bison make
 BuildRequires:  zlib-devel
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/parallel-libs/ptscotch/SPECS/ptscotch.spec]: https://github.com/openhpc/ohpc/pull/1678

File: components/parallel-libs/petsc/SPECS/petsc.spec [hunk 13]
```diff
@@ -38,7 +38,7 @@ BuildRequires:  valgrind-devel
 BuildRequires:  make
 BuildRequires:  xz
 BuildRequires:  zlib-devel
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/parallel-libs/petsc/SPECS/petsc.spec]: https://github.com/openhpc/ohpc/pull/1677

File: components/io-libs/sionlib/SPECS/sionlib.spec [hunk 14]
```diff
@@ -27,7 +27,7 @@ Source0:   http://apps.fz-juelich.de/jsc/sionlib/download.php?version=%{version}
 Patch0:    sionlib-llvm-arm.patch
 
 # For pre-processor only:
-%if 0%{?rhel}
+%if 0%{?rhel} ||  0%{?openEuler}
```
Comment (Yikun) [inline components/io-libs/sionlib/SPECS/sionlib.spec]: https://github.com/openhpc/ohpc/pull/1682

File: components/io-libs/netcdf-fortran/SPECS/netcdf-fortran.spec [hunk 15]
```diff
@@ -28,7 +28,7 @@ Source0:	https://github.com/Unidata/netcdf-fortran/archive/v%{version}.tar.gz
 
 BuildRequires:  zlib-devel >= 1.2.5
 BuildRequires:  libxml2-devel
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/io-libs/netcdf-fortran/SPECS/netcdf-fortran.spec]: https://github.com/openhpc/ohpc/pull/1681

File: components/distro-packages/python-Cython/SPECS/python-Cython.spec [hunk 16]
```diff
@@ -21,7 +21,7 @@ License:        Apache-2.0
 Group:          %{PROJ_NAME}/distro-packages
 Source0:        https://files.pythonhosted.org/packages/source/C/Cython/Cython-%{version}.tar.gz
 Source1:        python-Cython-rpmlintrc
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/distro-packages/python-Cython/SPECS/python-Cython.spec]: https://github.com/openhpc/ohpc/pull/1658

File: components/perf-tools/paraver/SPECS/paraver.spec [hunk 17]
```diff
@@ -27,7 +27,7 @@ Source0:       https://ftp.tools.bsc.es/wxparaver/wxparaver-%{version}-src.tar.b
 BuildRequires: gcc-c++
 BuildRequires: bison
 BuildRequires: boost-devel
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/perf-tools/paraver/SPECS/paraver.spec]: https://github.com/openhpc/ohpc/pull/1684

File: components/perf-tools/dimemas/SPECS/dimemas.spec [hunk 18]
```diff
@@ -28,7 +28,7 @@ Source0:	https://ftp.tools.bsc.es/dimemas/dimemas-%{version}-src.tar.bz2
 BuildRequires: boost-%{compiler_family}-%{mpi_family}%{PROJ_DELIM}
 Requires:      boost-%{compiler_family}-%{mpi_family}%{PROJ_DELIM}
 BuildRequires: bison make
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/perf-tools/dimemas/SPECS/dimemas.spec]: https://github.com/openhpc/ohpc/pull/1683

File: components/rms/munge/SPECS/munge.spec [hunk 19]
```diff
@@ -114,7 +114,7 @@ touch "$RPM_BUILD_ROOT"/%{_localstatedir}/run/munge/munged.pid
 %if 0%{?sles_version} || 0%{?suse_version}
 rm "$RPM_BUILD_ROOT"/etc/init.d/munge
 %endif
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/rms/munge/SPECS/munge.spec:117]: https://github.com/openhpc/ohpc/pull/1685

File: components/serial-libs/openblas/SPECS/openblas.spec [hunk 20]
```diff
@@ -29,6 +29,7 @@ Patch2:         openblas-noexecstack.patch
 Patch3:         fix-arm64-cpuid-return.patch
 ExclusiveArch:  %ix86 ia64 ppc ppc64 ppc64le x86_64 aarch64
 
+BuildRequires:  make ohpc-buildroot
```
Comment (Yikun) [inline components/serial-libs/openblas/SPECS/openblas.spec]: https://github.com/openhpc/ohpc/pull/1686

File: components/serial-libs/R/SPECS/R.spec [hunk 21]
```diff
@@ -44,14 +44,18 @@ BuildRequires:  xz-devel
 BuildRequires:  pcre2-devel
 BuildRequires:  libcurl-devel
 BuildRequires:  tk-devel
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/serial-libs/R/SPECS/R.spec]: https://github.com/openhpc/ohpc/pull/1687

File: components/serial-libs/scotch/SPECS/scotch.spec [hunk 22]
```diff
@@ -27,7 +27,7 @@ Source2:	%{pname}-rpmlintrc
 Patch0:		scotch-6.0.4-destdir.patch
 
 BuildRequires:	flex bison make
-%if 0%{?rhel}
+%if 0%{?rhel} || 0%{?openEuler}
```
Comment (Yikun) [inline components/serial-libs/scotch/SPECS/scotch.spec]: https://github.com/openhpc/ohpc/pull/1688

**Pull request closed without merging**